posit-dev/positron
# Settings: Improve titles

### Description
Addressess #1487

This is a first pass at improving the setting titles displayed in the settings table of contents for built in positron extensions. Any titles prefixed with "Positron" has been removed and ambiguity in extension titles have been addressed. The setting title in the TOC and category title for each setting item has been unified to match where possible. configuration keys have been changed to fix formatting of setting categories where possible.

Extension configuration keys that are prefixed with `positron` have been left as is for now (excluding theInterpreter settings) since the prefix does not effect the settings UI for now. We may want to address improving the key names for folks who interact with settings via the JSON file. It may be worth tackling after settling on a standard for configuration name formatting for bundled extensions.

The "Positron > Interpreters" settings now live under "Interpreters"/

The following extensions were modified and have the following setting titles in the TOC:
- open-remote-ssh: Remote SSH
- positron-notebook-controllers: Notebook Controllers
- positron-r: R
- positron-reticulate: Reticulate
- positron-run-app: App Launcher
- positron-supervisor: Kernel Supervisor
- positron-viewer: URL Viewer

### QA Notes

We'll want to verify there aren't any regressions with the settings for the extensions listed above, excluding any experimental feature settings.

For the interpreter settings we should verify that changing the automatic startup behavior, restart on crash behavior, and viewing sessions for interpreters are respected as usual. Some of this behavior should be captured by our e2e tests which will be run against this change before merging.

Note that as part of this change, the qa-example-content repo did need a change due to a setting key being renamed: https://github.com/posit-dev/qa-example-content/pull/41

### Screenshots

https://github.com/user-attachments/assets/fc66391a-5e0e-49a9-b717-a0d5401e2d1e
